### ISSUE TITLE
Cannot update to 1.19.3 (OLD: Update server software to 1.19.3)

### ISSUE THREAD

**Issue by WarpOverload** [TheRemote/Legendary-Java-Minecraft-Geyser-Floodgate#9]
See above, I restarted my server twice and it didn't automatically update! Any help would be greatly appreciated.

**Comment by WarpOverload** [comment]
I also tried `sudo docker pull 05jchambers/legendary-minecraft-geyser-floodgate:latest.` Nothing changed.

**Comment by WarpOverload** [comment]
Interactive terminal outputs: `[18:31:35 WARN]: [Geyser-Spigot] Your server software does not support the Java version that Geyser requires (1.19.3, 1.19.2). Please install ViaVersion (https://ci.viaversion.com/job/ViaVersion/) or update your server software!`

**Comment by TheRemote** [comment]
There's no Paper Minecraft 1.19.3 yet.  You can see this on Paper's site at https://papermc.io/downloads.

It probably won't be long before this is done as I see the release as only 9 hours ago.  You never have to wait for me to update the container though.  There's a version switch when you run the container you can use with -e Version=1.19.3.  It won't work if there's no Paper Minecraft server out for it to download yet though.

In fact if you pull the latest container you might have to use the Version switch to set it back to 1.19.2 before it can run.  Hopefully it won't be too long until that happens or it will mean I should have probably waited until it was released to even change it for the container.  I'm taking a leap of faith that this will happen within the next 24 hours or so hopefully.

These never update at the same time as vanilla though as it takes some time for the developers to catch up on the forked projects.  You can typically keep that Paper Minecraft download page up and keep refreshing until you see 1.19.3 available and at that point it should be good to go!

**Comment by WarpOverload** [comment]
I see... Thanks for the help though.

**Comment by TheRemote** [comment]
No problem at all!  Hopefully it won't be long here (it usually isn't too long).  Take care!

**Comment by ChrisTracy** [comment]
@TheRemote You are quick. Thanks for building this project and keeping it updated! 

So when paper releases a new version all should have to do is restart the server to update?

**Comment by TheRemote** [comment]
> @TheRemote You are quick. Thanks for building this project and keeping it updated!
> 
> So when paper releases a new version all should have to do is restart the server to update?

Thank you!  So the container version you are running now is probably defaulting to 1.19.2.  If you pull the new version now with:

`sudo docker pull 05jchambers/legendary-minecraft-geyser-floodgate:latest`

The default will change to 1.19.3.  This would be bad at the moment since it will basically fail to download the server since the download isn't up for Paper 1.19.3 yet.

No matter what you do though to override this you can launch the container with:

```
-e Version=1.19.3
-e Version=1.19.2
```

This means that once the Paper download is available you could actually stick with the container version you have without even updating and run the newer version if that makes sense.

That should give you a few options at least.  The easiest thing to do would probably just be to pull the new one once you see that it's available on Paper's web site and that will keep your default version sync'ed up with the latest version of Minecraft.

Either way though you do have an option to control this if things are working out weird / I'm slow to update / whatever it might be.  Hopefully that helps!

**Comment by erikpoole** [comment]
I may be missing something obvious here, but I'm not sure the environment variable completely fixes the issue.  I've attached two log files, one with the environment variable and one without, and I do see that the one with the version variable is able to download the correct version of paperclip while the one without is not.  

However, both seem to still be running minecraft server 1.19.2 and claim to be running Paper for 1.19.2 as well (despite maybe the failed download for the one without the version variable)?  With the environment variable set I'm still unable to connect to the server from bedrock clients and get the

```
[22:13:40 WARN]: [Geyser-Spigot] Your server software does not support the Java version that Geyser requires (1.19.3, 1.19.2). Please install ViaVersion (https://ci.viaversion.com/job/ViaVersion/) or update your server software!
```

warning.

[without-version-env-variable.txt](https://github.com/TheRemote/Legendary-Java-Minecraft-Geyser-Floodgate/files/10182437/without-version-env-variable.txt)
[with-version-env-variable.txt](https://github.com/TheRemote/Legendary-Java-Minecraft-Geyser-Floodgate/files/10182438/with-version-env-variable.txt)

**Comment by TheRemote** [comment]
There is no Paper 1.19.3 yet.  Go to https://papermc.io/downloads.  If there is no Paper 1.19.3 then there is nothing to download.  That is literally all it does.  It goes to that web site and downloads the latest version of the Paper Minecraft server available.

If you didn't use this script and tried to download Paper Minecraft 1.19.3 you would discover it's simply not released yet.  That is what is happening.  My environment variable can't make it exist.

You can set the environment variable and it will do nothing if there's nothing on the server for it to download.  It will just fail and if you already have an older version installed it runs that instead. 

In fact if people pull the new container now and don't have a server installed the server will not load at all and they would have to use the environment variable to go back to 1.19.2 until Paper posts a download of 1.19.3.  Hopefully that helps!

**Comment by erikpoole** [comment]
I apologize, I was unclear.  This is with me setting the environment variable to 1.19.2 to try and get my server back up again.  I've attached my compose.yml for reference.  
[compose.yml](https://github.com/TheRemote/Legendary-Java-Minecraft-Geyser-Floodgate/files/10182466/compose.txt)

**Comment by TheRemote** [comment]
No worries at all, I will take a look here real quick!

**Comment by TheRemote** [comment]
Oh I think I understand what it's saying.  My guess is what is happening is that the Geyser clients are probably connecting with 1.19.3 (or what would be the equivalent of it on Bedrock).

All the Geyser changes are here:  https://ci.opencollab.dev/job/GeyserMC/job/Geyser/job/master/

They're making a lot of updates to Geyser.  The last change was 2 hours ago.  It looks like they are actively working on this.

The way the container works is that whenever you restart the container it will check if there's a Geyser update, a Floodgate update and a Paper update from their web sites.  Essentially whenever you see a new version on that site restarting the container will update it for you if that makes sense.

This message tells me that something still needs to be updated and it looks like they're in the middle of fixing it to me.  I'd keep an eye on the Geyser build changes and as you see them push out a new build try restarting the container.  I think this is probably in the middle of being fixed!

**Comment by erikpoole** [comment]
Thanks for the context!  I really appreciate blazing fast responses.

**Comment by TheRemote** [comment]
No problem at all, thanks for sending me a tip as well, I just saw that!

I'm honestly so confused by their warning message and it still bothers me looking at it:

`Your server software does not support the Java version that Geyser requires (1.19.3, 1.19.2)`

That is a horrible error message.  Why does it say it doesn't support the "Java" version and then list two Minecraft server versions 1.19.2 and 1.19.3?  Those are not Java versions.

The Java version we are using is OpenJDK 19 though.  The version number for that?  19.0.1.  Ugh.  The container has been using this for quite some time though so I don't expect this to be actually Java version related despite the error message.  The version numbers they are giving clearly seem to be Minecraft versions rather than Java versions.

I will keep a close eye on this and I'm hoping that when I wake up tomorrow it will already be patched and fixed.  Theoretically if it is OpenJDK related I could drop it back down to OpenJDK 18 but this would make no sense.  OpenJDK 19 is the current version of OpenJDK as of 2 months ago or so.  It would make no sense for the Geyser devs to remove support for the current version now and I really think Geyser must just need some more tweaks here for the new version (particularly to handling then new Bedrock clients it seems).

**Comment by TheRemote** [comment]
For an update on this it looks like Geyser did do an update but all they did was change the error messages according to their patch logs.  Looking back through the Geyser changes it looks like they dropped support for some Bedrock versions about a week or so ago.

Spigot 1.19.3 has been released so Paper can't be too far behind (Paper is downstream from Spigot).  I think that the Paper 1.19.3 update will solve this as from what I can tell the message is supposed to mean that you need to be running 1.19.3 for the Bedrock clients to connect.  The message might be more clear about this in more recent Geyser versions but I think the Paper update is what will fix this once and for all!

**Comment by WarpOverload** [comment]
@TheRemote I see Paper 1.19.3 is out, (for quite while now, actually) I restarted the server and when that didn't work, I tried the pull command. It hasn't updated to the latest build, and is stuck on 1.19.2, paper version #307. Interactive terminal outputs:
`
Updating to most recent paperclip version ...
Latest paperclip build found: 307
`
Any ideas?

**Comment by WarpOverload** [comment]
I was not able to do this earlier because I was away from home and forgot to port forward SSH.

**Comment by TheRemote** [comment]
This is all closed out and complete.  If you are having trouble updating please remove the container and pull it fresh.

The update is completed.  You can also use the -e Version=1.19.3 flag.  You do not need to update the container to update to the new version.

If you are having trouble try removing all container instances and pull again.  There's really nothing I can do to change your containers version if you have pulled the latest.  The updated container is out there or override it with the version flag I would suggest!

The problem is that you are not running the latest version.  You said you have pulled it.  Some explanations for why it could still be running 1.19.2 are you started it with the -e Version=1.19.2 flag before or it's not actually pulling successfully.

Either way just remove whatever you have in there.  Do docker container ls and then do docker container rm to all of those containers.  Then run the latest version.

It is not required for you to update the container or pull the latest to do this. There were no changes.  All I did was change the version from the 1.19.2 to 1.19.3.  There is literally not a single byte different in the latest pulled container and this one other than it just defaults to 1.19.3.  

 Just start it with -e Version=1.19.3.  I've covered how to do this from scratch multiple times in this thread so I'm not sure what to tell you other than you aren't running the latest version and should remove everything and pull it fresh!

If you'd like me to support this further can you please share your Docker startup line with us?  It should definitely have a -e Version=1.19.3 on it.  If it does not then that is going to be the answer and it would have fixed it since about 36 hours from when 1.19.3 was officially released (as that is when Paper 1.19.3 came out).
